src/RRTStar/obstacles.cpp

```diff
@@ -0,0 +1,74 @@
+#include "obstacles.h"
+
+#include <iostream>
+
+Obstacles::Obstacles()
+{
+}
+
+/**
+ * @brief Obstacles are stored as rectangles. Rectangle is denoted by two points : topLeft and bottomRight.
+ * @param firstPoint
+ * @param secondPoint
+ */
+void Obstacles::addObstacle(Vector2f firstPoint, Vector2f secondPoint)
+{
+    // Get topLeft and bottomRight points from the given points.
+    Vector2f tmp;
+    if (firstPoint.x() > secondPoint.x() && firstPoint.y() > secondPoint.y()) {
+        tmp = firstPoint;
+        firstPoint = secondPoint;
+        secondPoint = tmp;
+    } else if (firstPoint.x() < secondPoint.x() && firstPoint.y() > secondPoint.y()) {
+        int height = firstPoint.y() - secondPoint.y();
+        firstPoint.y() -= height;
+        secondPoint.y() += height;
+    } else if (firstPoint.x() > secondPoint.x() && firstPoint.y() < secondPoint.y()) {
+        int length = firstPoint.x() - secondPoint.x();
+        firstPoint.x() -= length;
+        secondPoint.x() += length;
+    }
+    firstPoint.x() -= BOT_CLEARANCE;
+    firstPoint.y() -= BOT_CLEARANCE;
+    secondPoint.x() += BOT_CLEARANCE;
+    secondPoint.y() += BOT_CLEARANCE;
+    obstacles.push_back(make_pair(firstPoint, secondPoint));
+}
+
+/**
+ * @brief Check if a line segment intersects a rectangle.
+ * @param p1
+ * @param p2
+ * @return
+ */
+bool Obstacles::isSegmentInObstacle(Vector2f &p1, Vector2f &p2)
```

Comment:
IMHO, it would be nicer if we had an Obstacle class, which would contain most of the logic from addObstacle in the constructor, and would provide a IsSegmentInObstacle() method. Thus, you wouldn't have to store pairs of points if vectors, and would be able to directly store a vector of obstacles in rrt.cpp. As it stands, obstacles is somewhere in between a collection class and a class that implements an entity. What do you think?

Reply:
I agree, this is better from implementation standpoint. I have added these changes :)